src/exes/moc.ml

```diff
@@ -192,7 +192,11 @@ let process_files files : unit =
     end;
 
     if !stable_types then begin
-      let sig_file = Filename.remove_extension (Filename.basename !out_file) ^ ".sig" in
+      let sig_file =
+          (if Filename.extension !out_file <> "" then
+             Filename.remove_extension !out_file
+           else !out_file)
```

Comment:
Isn't `remove_extension` already doing the same conditional?

Reply:
I think the point was that if out file doesn't have an extension we append .sig otherwise replace the extension with .sig.
Otherwise we overwrite and out file we just produced.

Reply:
```
[nix-shell:~/motoko/test/run-drun]$ moc -o foo.wasm --stable-types stable-sig.mo 
[nix-shell:~/motoko/test/run-drun]$ ls foo*
foo.most  foo.wasm

[nix-shell:~/motoko/test/run-drun]$ moc -o bar --stable-types stable-sig.mo

[nix-shell:~/motoko/test/run-drun]$ ls bar*
bar  bar.most


```

Reply:
Yes, but `remove_extension` already is the identity when there is no extension, so checking with a conditional is redundant.

Reply:
Done

Reply:
(in full pr)